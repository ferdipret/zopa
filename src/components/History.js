import React from "react"
import styled from "styled-components"

function History({ transactions }) {
  return transactions.map((t, i) => (
    <div key={i}>
      <div>{t.name}</div>
      <div>{t.email}</div>
      <div>{t.amount}</div>
    </div>
  ))
}

export default History
