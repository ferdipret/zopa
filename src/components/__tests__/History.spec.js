import React from "react"
import { shallow } from "enzyme"
import History from "../History"

describe("History", () => {
  const transactions = [
    {
      name: "Jimmy",
      email: "jimmyblogs@zopa.com",
      amount: 2000
    }
  ]

  const wrapper = shallow(<History transactions={transactions} />)
  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
