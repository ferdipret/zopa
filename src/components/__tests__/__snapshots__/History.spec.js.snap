// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <History
    transactions={
        Array [
            Object {
              "amount": 2000,
              "email": "jimmyblogs@zopa.com",
              "name": "Jimmy",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div>
      <div>
            Jimmy
      </div>
      <div>
            jimmyblogs@zopa.com
      </div>
      <div>
            2000
      </div>
</div>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div>
        <div>
                Jimmy
        </div>
        <div>
                jimmyblogs@zopa.com
        </div>
        <div>
                2000
        </div>
</div>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
